### DISCUSSION TITLE
Deploying Career Portal with Advanced Hosting to Firebase

### DISCUSSION THREAD

**Discussion by inTulsa** [bullhorn/career-portal#412] [Q&A] [answered]
I'm trying to deploy the career portal with advanced hosting to Firebase. When testing locally, I get this error:
_Error: No NgModule metadata found for '[object Object]'_

## Current behavior
Advanced Hosting function not working when deploying to Firebase.

It's difficult to find documentation regarding Bullhorn Career Portal and Advanced Hosting options. Can you provide me with a recommended method for advanced hosting that would work well with an Angular 7 Universal app?

## Steps to replicate behavior

1. Build the dist/browser and dist/server files
2. Initialize Firebase and set public directory to dist/browser
3. Move the function to dist
4. Edit the server.ts to import firebase-functions and add an entry point.
5. "Firebase Serve" or "Firebase Deploy"

## Screenshots

**Error:**
![image](https://user-images.githubusercontent.com/84162344/127075791-901a0743-9879-4e83-a033-d2d05df55e2c.png)

**Server.ts:**
![image](https://user-images.githubusercontent.com/84162344/127076263-15ed9a82-628e-48a8-ba46-5207ad74cd15.png)

**Comment by charlesabarnes** [comment, marked as answer]
Hello, to answer your question Heroku works using just the built-in deploy template.  I know EC2 works as well using PM2. 

https://heroku.com/deploy?template=https://github.com/bullhorn/career-portal

**Reply by inTulsa** [reply]
This is all I needed! Thank you for the quick response!
